src/allocator.rs

```diff
@@ -288,8 +300,26 @@ impl Block {
         md.drop_vptr = (value as u64).into();
         self.header_mut().set_metadata(md);
     }
+
+    pub(crate) unsafe fn drop(self) {
+        assert!(
```

Comment:
I'm not totally sure I understand how/why this `assert` might be called.

Reply:
Drop has a precondition that can only be called on allocation blocks which are managed by the GC. Since we've now moved `drop` to be a method call on `Block` instead of directly inside the `enter_drop_phase` method in the collector, it is theoretically possible to construct a `Block` from a base pointer to a non-garbage-collected allocation block. This assert acts as a sanity check against that.

Reply:
Can the end user ever do this or is it an internal sanity check? If the latter, should it be `debug_assert`?

Reply:
It's an internal sanity check, the end user can never drop a `Gc` value explicitly. I'm happy to make it a `debug_assert`.

Reply:
OK, let's do that then.

Reply:
Fixed: 966631df121a9fda0f79e7d189bc9ede62c29c51